lib/src/main/java/graphql/nadel/validation/NadelFieldValidation.kt

```diff
@@ -181,6 +180,6 @@ internal class NadelFieldValidation(
 
     context(NadelValidationContext)
     private fun isCombinedType(type: GraphQLNamedSchemaElement): Boolean {
-        return isCombinedType(engineSchema, type)
+        return type.name in combinedTypeNames
```

Comment:
This is run on _every_ field.

The old `isCombinedType` would iterate through all root fields, which was very slow.

This procomputes it and puts that information in the NadelValidationContext so that we only iterate through the root fields once.